apps/authentication-service/src/shared/auth/authService.ts

```diff
@@ -0,0 +1,112 @@
+import fs from "fs";
+import path from "path";
+import os from "os";
+import { TokenSet } from "auth0";
+import { decode, JwtPayload } from "jsonwebtoken";
+
+import envs from "@/config";
+import { IAuthService, ICredentialRequestBody } from "@/shared/auth/authModel";
+import { ISingleton } from "@/shared/models";
+
+export const authEndpoint = "oauth/token";
+
+export const AuthService: ISingleton<IAuthService> = (() => {
+  let instance: IAuthService;
+
+  const createInstance = (): IAuthService => {
+    const domain: string = envs.AUTH0_DOMAIN;
+    const audience: string = envs.AUTH0_AUDIENCE;
+    const clientId: string = envs.AUTH0_CLIENT_ID;
+    const clientSecret: string = envs.AUTH0_CLIENT_SECRET;
+    const tokenFileName: string = envs.TOKEN_CACHE_FILE;
+
+    let token: string;
+    const tokenFile: string = path.join(os.tmpdir(), tokenFileName);
+
+    const setCachedToken = (token: string): void => {
+      try {
+        fs.writeFileSync(tokenFile, token);
+      } catch (Error) {
+        console.log("Unable to cache token", Error);
+      }
+    };
+
+    const getCachedToken = (): string | undefined => {
```

Comment:
Why do we need to store the token in a file? Why don't we keep it in memory?

Reply:
It is kept in memory. This allows us to use a valid token for it's entire lifetime, instead of getting a new token anytime the service restarts. This is also a way to keep us from burning through our monthly allotment of access tokens. I found this idea in the Auth0 community, from an employee of Auth0.

Once I implemented this caching, I went from using over 100 per day, to less than 20. 
<img width="834" alt="image" src="https://github.com/Universal-Connect-Project/ucp-hosted-app/assets/1331818/47f323b6-01f6-450b-8f9e-237f5061cfc9">

Reply:
As we discussed in person it looks like the file is mostly used for local development to ensure that restarting the service won't request a new token.